challenge-11/challenge-11.js

```diff
@@ -1,54 +1,94 @@
-// Envolva todo o código desse arquivo em uma IIFE (incluindo esse comentário).
+(function(){
+    // Envolva todo o código desse arquivo em uma IIFE (incluindo esse comentário).
+    /*
+    Crie uma variável chamada `once`, que recebe `false` como seu valor.
+    Crie um loop que execute enquanto essa variável for verdadeira. Dentro do
+    loop, mostre no console a mensagem:
+    'Entrou ao menos uma vez!'
+    Qual loop você deve usar para que essa mensagem seja mostrada no console?
+    */
+    var once = false;
+    do {
+        console.log('Entrou ao menos uma vez!');
+    } while( once === true );
 
-/*
-Crie uma variável chamada `once`, que recebe `false` como seu valor.
-Crie um loop que execute enquanto essa variável for verdadeira. Dentro do
-loop, mostre no console a mensagem:
-'Entrou ao menos uma vez!'
-Qual loop você deve usar para que essa mensagem seja mostrada no console?
-*/
-// ?
+    /*
+    Crie um objeto chamado `person`, que receba as seguintes propriedades:
+    - 'name', 'age', 'weight' e 'birthday'. Preencha com os valores corretos
+    para o nome, idade, peso e data de nascimento dessa pessoa.
+    */
+    var person = {
+        name: 'Clara',
+        age: 20,
+        weight: 60,
+        birthday: '17-03-1995'
+    };
 
-/*
-Crie um objeto chamado `person`, que receba as seguintes propriedades:
-- 'name', 'age', 'weight' e 'birthday'. Preencha com os valores corretos
-para o nome, idade, peso e data de nascimento dessa pessoa.
-*/
-// ?
+    /*
+    Use um loop para percorrer o objeto criado acima, mostrando no console
+    a frase:
+    'The [PROPERTY] of person is [VALUE]'
+    Aproveite e crie uma variável `counter` que vai contar quantas propriedades
+    esse objeto tem.
+    Após o loop, mostre a frase:
+    'The person has [COUNTER] properties'
+    */
+    var counter = 0;
+    for ( var prop in person ){
+        console.log( 'The ' + prop + ' of person is ' + person[prop] );
+        counter++;
+        console.log( 'The person has ' + counter + ' properties.' );
```

Comment:
-1
Essa frase deve ser mostrada fora do loop:

> Após o loop, mostre a frase:
'The person has [COUNTER] properties'